smdebug/tensorflow/keras.py

```diff
@@ -122,9 +124,12 @@ def __init__(
         self.has_logged_unsupported_tensors_in_non_eager_execution = False
         self.prepared_tf2_collections = False
         self.prepared_gradient_tape_collections = False
+        # this flag indicates whether profiling is enabled for native TF2 training
+        self.is_profiler_enabled_for_native_training = False
+        self.closed = False
 
-        if python_profiler:
-            atexit.register(python_profiler.stop_profiling, StepPhase.END)
+        if self.python_profiler:
+            atexit.register(self.profiling_end)
```

Comment:
At the end of program, will this object (self) be still valid?  What happens if it goes out of scope?

Reply:
It's still valid. `self.profiling_end` is passed as an argument like any other `callable` in Python, so it can directly be called without `self` needing to be in scope. 

As verification that this works, the end to end SageMaker tests pass (with the post-hook-close Python profiling stats being successfully collected for the jobs).